public/javascripts/discover.js

```diff
@@ -1,11 +1,20 @@
 var discover = (function() {
-    var inLibrary = m.prop(currentInLibrary)
-    var displayedSongs = m.prop(songs)
-    var initial = 10
-    var langShown = m.prop('all')
-    var langFilter = m.prop([])
-    var searchString = m.prop()
-    var addButtonDOM = m.prop()
+    var inLibrary = m.prop(currentInLibrary);
+    var displayedSongs = m.prop(songs);
+    var initial = 10;
+
+    //langShown store the language id. it is for showing the songs title in 'langShown'
+    var langShown = m.prop('all');
+
+    //langShownLabel stores the language label such as 'English', 'Spanish', etc
+    var langShownLabel = m.prop();
```

Comment:
you could combine langShown and langShownLabel into an object. <code>m.prop({label: 'All', code: 'all'})</code>